### PR TITLE
Cast to bool boolean key in the ini file. Fixes #389

### LINKED ISSUE #389: Invalid QgsSettings cause the plugin to not be able to run any algorithm
I have somehow managed to save some incorrect state into my `QgsSettings` and now the plugin is unable to run its algorithms. I am attaching a copy of the `.ini` file saved by QGIS in order to help debugging (had to put it inside a zip in order to trick github).

If I try to run a TE algorithm with these QSettings in place my QGIS just freezes and I eventually am forced to kill it.

When a new QGIS profile is used, one with also a new QgsSettings object, the algorithm execution error does not manifest - If I start a new QGIS profile, install TE into it and try to run an algorithm, execution fails in a wel behaved manner, showing me a pop up dialogue with the following message:

```
Choose an area of interest.
```

So it seems that the original `QgsSettings` have some error in the Trends.Earth region of interest.

QGIS settings file: [qgis-settings-ini.zip](https://github.com/ConservationInternational/trends.earth/files/6310150/qgis-settings-ini.zip)

### PR DESCRIPTION
**Pull request by luipir**
fixes #389

### CODE REVIEW

File: LDMP/settings.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def __init__(self, parent=None):
 
     def load_settings(self):
 
-        buffer_checked = self.settings.value("trends_earth/region_of_interest/buffer_checked", False) == 'True'
+        buffer_checked = self.settings.value("trends_earth/region_of_interest/buffer_checked", False, type=bool) == 'True'
```
Comment (ricardogsilva) [inline LDMP/settings.py:276]: This line is flawed, you are comparing against a string but at the same time you cast the output of `settings.value` to a `bool`. This test is always false. I suggest this instead:

```suggestion
        buffer_checked = self.settings.value("trends_earth/region_of_interest/buffer_checked", False, type=bool)
```

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
Nice work finding the root of the problem :smile_cat: 

Please fix the faulty check in the inline comment and I'll be :+1: to merge

File: LDMP/settings.py (hunk 1 above)
Comment (luipir) [reply LDMP/settings.py:276]: good catch :)... I'll add a modification to a similar code in the same PR

**Comment by luipir** [comment]
@ricardogsilva please review again... I removed all compare to 'True? all around the code (damn I've all the monitor flickering)

**Review by ricardogsilva — APPROVED** [review]
LGTM!

**Pull request merged**